### PR TITLE
Avoid cloning LocalDecls

### PR DESCRIPTION
**Pull request by ptrojahn**

### CODE REVIEW

**Comment by JohnTitor** [comment]
In case this is perf-sensitive,
@bors try @rust-timer queue

**Comment by JohnTitor** [comment]
Looks like it's perf-insensitive.
@bors rollup=maybe

r? @eddyb as FIXME is yours.

**Comment by jackh726** [comment]
@ptrojahn can you rebase? I'll take the review here. r? @jackh726

**Comment by ptrojahn** [comment]
Didn't see the conflict. Rebased

File: compiler/rustc_mir/src/transform/inline.rs [hunk 1]
```diff
@@ -607,13 +607,9 @@ impl Inliner<'tcx> {
...
-                caller_body.local_decls.extend(
-                    // FIXME(eddyb) make `Range<Local>` iterable so that we can use
-                    // `callee_body.local_decls.drain(callee_body.vars_and_temps())`
-                    callee_body
-                        .vars_and_temps_iter()
-                        .map(|local| callee_body.local_decls[local].clone()),
-                );
+                caller_body
+                    .local_decls
+                    .extend(callee_body.local_decls.drain(callee_body.arg_count + 1..)); // vars and temps
```
Comment (jackh726) [inline compiler/rustc_mir/src/transform/inline.rs:612]: So, I don't think "exposing" the internal logic of `arg_count + 1..` is all that great. Could you add a new function on `Body`, `drain_vars_and_temps`?

**Review by jackh726 — CHANGES_REQUESTED** [review]

File: compiler/rustc_mir/src/transform/inline.rs (hunk 1 above)
Comment (ptrojahn) [reply compiler/rustc_mir/src/transform/inline.rs]: Agreed

**Pull request merged**